### PR TITLE
feature/harfanglab_add_field_connection_successful

### PR DESCRIPTION
**Pull request by clement-burtscher-sekoia**
## Description

Relates to https://github.com/SekoiaLab/integration/issues/1875

### Fixed

- Generalize `connection_successful` handling with a dedicated stage so it is parsed whenever present in raw logs, regardless of event dataset (`network`, `connectionlog`, etc.)
- Update `event.outcome` from `connection_successful` in the same dedicated stage for consistent behavior across datasets

## Type of Change

- [ ] New intake format
- [x] Update to existing intake format
- [x] Bug fix
- [ ] Documentation update
- [ ] Other (please describe):

## Affected Intake Formats

- Vendor/product: HarfangLab/harfanglab

### 🔒 CRITICAL - Security Requirements (MUST BE VERIFIED FIRST)

- [ ] **No real email addresses** in test files (use: user@example.com, test@example.org)
- [ ] **No real passwords or API keys** (use: "P@ssw0rd!", "xxxxxxxxxxxx", "FAKE_TOKEN_123")
- [ ] **No real IP addresses** (use TEST-NET ranges: 192.0.2.x, 198.51.100.x, 203.0.113.x)
- [ ] **No real phone numbers** (use: +1-555-0100 or similar test numbers)
- [ ] **No real names or PII** (use: "John Doe", "Jane Smith", "User123")
- [ ] **No real company/organization names** in test data (unless public vendors)
- [ ] **No real usernames, hostnames, or domain names** (use: user1, host.example.com, test.corp)
- [ ] All test data has been anonymized and verified

### Required for All PRs

- [ ] Code has been linted
    - [ ] with the linter for manifest, smart-descriptions, and test files (`uv run python linter.py check --changes` in utils/)
    - [ ] with Prettier for the parser code files (`npx prettier --write <module>/<format>/ingest/parser.yml`)
- [ ] All CI/CD checks pass
- [ ] Changes follow the contribution guidelines in `CONTRIBUTING.md`

### Required for New/Updated Intake Formats

- [ ] Clear descriptions provided for modules and formats
- [ ] Logo files included for new modules/formats
- [ ] Parser test coverage is at least 75%
- [ ] At least one smart-description is provided
- [ ] README.md updated (if applicable)
- [ ] Taxonomy mappings are correct and documented
- [ ] Test cases cover edge cases and error handling

### Testing

- [ ] Local tests pass (`uv run pytest` in utils/)
- [ ] Sample data is representative and **completely anonymized**
- [ ] **No sensitive information** (PII, credentials, real IPs, real emails) in test files
- [ ] Test data uses example.com/example.org domains and TEST-NET IP ranges
- [ ] Parser handles malformed data gracefully

### Documentation

- [ ] Code changes are documented
- [ ] Smart-descriptions are clear and informative
- [ ] Field mappings are explained

## Related Issues

Closes #

## Summary by Sourcery

Generalize handling of the HarfangLab `connection_successful` field across all datasets and align `event.outcome` mapping accordingly.

Bug Fixes:
- Ensure `connection_successful` is parsed and mapped consistently to `event.outcome` regardless of the originating event dataset.

Enhancements:
- Introduce a dedicated `connection_successful_info` pipeline stage to centralize parsing and mapping of connection success status.
- Update tests and samples to cover the generalized `connection_successful` handling.
- Bump the HarfangLab intake version to 1.3.5 with corresponding changelog entries.

### CODE REVIEW

File: HarfangLab/harfanglab/ingest/parser.yml [hunk 1]
```diff
@@ -299,6 +302,20 @@ stages:
...
+          harfanglab.connection_successful: "{{json_event.message.connection_successful in [true, 'true', 'True', 1, '1']}}"
+        filter: "{{json_event.message.connection_successful in [true, 'true', 'True', 1, '1', false, 'false', 'False', 0, '0']}}"
+
+      - set:
+          event.outcome: "success"
+        filter: "{{json_event.message.connection_successful in [true, 'true', 'True', 1, '1']}}"
+
+      - set:
+          event.outcome: "failure"
+        filter: "{{json_event.message.connection_successful in [false, 'false', 'False', 0, '0']}}"
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply HarfangLab/harfanglab/ingest/parser.yml]: Fixed with commit [Apply changes requested by Copilot](https://github.com/SEKOIA-IO/intake-formats/pull/2802/changes/355a8f2f7edd796c86330ae521f30828fdc93f2d)

File: HarfangLab/harfanglab/tests/network8.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
       ],
       "code": "8194",
       "dataset": "network",
+      "outcome": "success",
       "type": [
         "connection"
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply HarfangLab/harfanglab/tests/network8.json:15]: Fixed with commit [Apply changes requested by Copilot](https://github.com/SEKOIA-IO/intake-formats/pull/2802/changes/355a8f2f7edd796c86330ae521f30828fdc93f2d)

File: HarfangLab/harfanglab/tests/network7.json [hunk 3]
```diff
@@ -10,6 +10,7 @@
       ],
       "code": "8194",
       "dataset": "network",
+      "outcome": "success",
       "type": [
         "connection"
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply HarfangLab/harfanglab/tests/network7.json:15]: Fixed with commit [Apply changes requested by Copilot](https://github.com/SEKOIA-IO/intake-formats/pull/2802/changes/355a8f2f7edd796c86330ae521f30828fdc93f2d)

**Review by mchupeau-sk — APPROVED** [review]
LGFM
